tests/canary_test.rs

```diff
@@ -1,11 +1,42 @@
+use std::sync::Arc;
+
 use axum_test::TestServer;
-use fms_guardrails_orchestr8::server::get_health_app;
+use fms_guardrails_orchestr8::{
+    config::OrchestratorConfig,
+    orchestrator::Orchestrator,
+    server::{get_health_app, ServerState},
+};
 use hyper::StatusCode;
+use serde_json::Value;
+use tokio::sync::OnceCell;
+use tracing::debug;
+use tracing_test::traced_test;
+
+/// Async lazy initialization of shared state using tokio::sync::OnceCell
+static ONCE: OnceCell<Arc<ServerState>> = OnceCell::const_new();
+
+/// The actual async function that initializes the shared state if not already initialized
+async fn shared_state() -> Arc<ServerState> {
+    let config = OrchestratorConfig::load("config/test.config.yaml")
+        .await
+        .unwrap();
+    let orchestrator = Orchestrator::new(config).await.unwrap();
+    Arc::new(ServerState::new(orchestrator))
+}
 
 /// Checks if the health endpoint is working
+/// NOTE: We do not currently mock client services yet, so this test is
+/// superficially testing the client health endpoints on the orchestrator is accessible
+/// and when the orchestrator is running (healthy) all the health endpoints return 200 OK.
+/// This will happen even if the client services or their health endpoints are not found.
+#[traced_test]
```

Comment:
what is the need for traced_test (and adding this library) ?

Reply:
This was definitely not a strong need by any means so I'll let you have the final say on whether we want it or not. It just enabled testing to use tracing but in retrospect the additional value is perhaps too minimal. Let me know if you want to use this (so we can use `debug!`) or if you prefer to just `println!`. I originally added this as I was trying to decide between leaving the `println!`s in the PR or not, as I found them useful for debugging but they require some additional work (json parsing, pretty printing) that made it feel worth it to leave the `println!`/`debug!`s behind so that someone debugging in the future doesn't need to re-figure out how to pretty print the response. And then I thought if I'm leaving behind debug prints, maybe I should make them explicitly `debug!` with the potential additional benefit of having log level configurable.

If you do want to keep `tracing-test` around, I will move it to dev-dependencies, but I also don't need any convincing to just remove it.

Reply:
make sense. we should not use println, so lets keep `tracing-test` and move it to `dev-dependencies`. Thanks